### DISCUSSION TITLE
FHIR bundle in JSON

### DISCUSSION THREAD

**Discussion by testlorenzospa** [FirelyTeam/firely-net-sdk#2503] [Q&A] [answered]
Hello.

I want to use this library to read a FHIR bundle in JSON from a RESTFul service.
However, I haven't a connection to use for this.
Can I read a fixed JSON string to verify how this library work? If yes, there is some example?

Let me know, please.

Thanks.

**Comment by mbaltus** [comment, marked as answer]
Yes, you can read a string from file or have an in-memory string and parse it to a Bundle object first, before you do what you would like after the REST call. Another option is to use one of the publicly available FHIR test servers.

Here's an example reading a Bundle from file and parsing it:
``` csharp
using Hl7.Fhir.ElementModel;
using Hl7.Fhir.Model;
using Hl7.Fhir.Serialization;
using System.Text.Json;

void Test() {
  JsonSerializerOptions options = new JsonSerializerOptions().ForFhir();

  string jsonInput = File.ReadAllText("bundle.json");
  Bundle bundle = JsonSerializer.Deserialize<Bundle>(jsonInput, options);

  foreach (var entry in bundle.Entry)
  {  // do something with the entry }
}
```

or use a test server:
``` csharp
using Hl7.Fhir.Model;
using Hl7.Fhir.Rest;

void Test() {
  FhirClient client = new FhirClient("https://server.fire.ly/R4");
  SearchParams searchParams = new SearchParams().Where("name=Everyman");
  Bundle searchResult = client.Search<Patient>(searchParams);

  foreach (var entry in searchResult.Entry)
  {  // do something with the entry }
}
```
